src/Uecode/Bundle/ApiKeyBundle/Security/Authentication/Provider/ApiKeyProvider.php

```diff
@@ -58,12 +66,8 @@ public function authenticate(TokenInterface $token)
      * @return bool|ApiKeyUserToken
      * @throws AuthenticationException
      */
-    protected function doAuth(UserProviderInterface $provider, TokenInterface $token)
+    protected function doAuth($provider, TokenInterface $token)
     {
-        if (!method_exists($provider, 'loadUserByApiKey')) {
```

Comment:
I'm checking constructor that $provider implements required interface, but I really missed the case someon use chain of providers.

I think it will be better to use instanceof instead of method_exists, will add it today.